src/front/wgsl/mod.rs

```diff
@@ -1813,16 +1815,37 @@ impl Parser {
                 let mut block = Vec::new();
 
                 lexer.expect(Token::Paren('('))?;
-                let initialization =
-                    self.parse_statement(lexer, context.reborrow(), is_uniform_control_flow)?;
-                let condition = self.parse_general_expression(lexer, context.as_expression())?;
+                let initialization = if lexer.peek() != Token::Separator(';') {
```

Comment:
we shouldn't ever use the `peek()` followed by `next()`. Instead, you can do `let initialization = if lexer.skip(..) { None } else { ... parse_statement() };`